Tribler/Core/APIImplementation/LaunchManyCore.py

```diff
@@ -761,10 +761,18 @@ def network_shutdown(self):
 
     def save_download_pstate(self, infohash, pstate):
         """ Called by network thread """
+
+        # if pstate can't be found
+        if not pstate.get('state', 'engineresumedata'):
```

Comment:
So, how about `save_download_pstate` always call `save_resume_data` and then set the `pstate_for_restart`. And then when receiving an alert, we actually write that one.